zhaquirks/orvibo/switch.py

```diff
@@ -0,0 +1,62 @@
+"""
+ORVIBO switches.
+"""
+
+from zigpy.profiles import zha
+from zigpy.quirks import CustomDevice
+from zigpy.zcl.clusters.general import (
+    Basic,
+    Groups,
+    Scenes,
+    OnOff,
+)
+
+from . import ORVIBO
+from ..const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+
+
+class T10W1ZW(CustomDevice):
+    """T10W1ZW in-wall switch"""
+
+    signature = {
+        # <SimpleDescriptor endpoint=1 profile=260 device_type=0
+        # device_version=0
+        # input_clusters=[0, 4, 5, 6]
+        # output_clusters=[0]>
+        MODELS_INFO: [(ORVIBO, "545df2981b704114945f6df1c780515a")],
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    Groups.cluster_id,
+                    Scenes.cluster_id,
+                    OnOff.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [Basic.cluster_id],
+            }
+        },
+    }
+
+    replacement = {
+        ENDPOINTS: {
+            1: {
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_LIGHT,
```

Comment:
why are you changing the device type?

Reply:
The dimmable one above makes sense. This one shouldn't be changed. The device is an in wall switch not a light. Please remove this. You can override the platform in HA if you want it to be a light. If this was the only change you can remove the entire quirk. @Adminiuga do you feel any different about this?

Reply:
Strictly not necessary, but I use it as a in-wall light switch for lights. I just re-used what I had done for deconz/Phoscon.  If it's an issue, we can drop this quirk.

Reply:
I would remove this as other users may not want this. This can be overridden w/ configuration for the device on a per user basis in configuration.yaml in HA.

Reply:
Sure, no problem. I'll make the change.

Reply:
One more thing though, before I delete this: I noticed that if I use the physical button to turn the light on/off the state is not updated in HA. Is this something that has to be added via the quirk? Turning the switch on/off via HA works fine.  Same behavior with the dimmer: changes with the physical button are not picked up by HA...

Reply:
Ugh, this could be 2 things: they didn’t pair correctly and they aren’t reporting changes as a result or they just flat out don’t report at all... and may need to be polled.

Reply:
Ok, I'll try to investigate. For now, I'll remove the quirk.